external/mbedtls/alt/pk_wrap_alt.c

```diff
@@ -846,7 +847,7 @@ int eckey_verify_wrap(void *ctx, mbedtls_md_type_t md_alg, const unsigned char *
 
 FREE:
 	sl_deinit(shnd);
-	if (ret != SECLINK_OK || hres != HAL_SUCCESS) {
+	if (ret != SECLINK_OK || hres != HAL_SUCCESS || hhres != HAL_SUCCESS) {
```

Comment:
Is it okay without defining hhres such as Line:748?

Reply:
Basically this is happened each of function in seclink always return SECLINK_OK.
So param hal_result_e should be checked all the time.
anyway each of step doesn't have a any dependency, and it doesn't jump to anywhere, so I think below will work.
```
ret = sl_ecdsa_verify_md(shnd, ecdsa_mode, &t_hash, &t_sig, key_idx, &hres);
if (ret != SECLINK_OK) { // but this will never be happened
	ret = MBEDTLS_ERR_RSA_UNSUPPORTED_OPERATION;
}

if (hres != HAL_SUCESS) {
	ret = MBEDTLS_ERR_RSA_UNSUPPORTED_OPERATION;
}

free(t_hash.data);
free(t_sig.data);

FREE_WITH_KEY:
	(void)sl_remove_key(shnd, key_type, key_idx, &hres);
	if (hres != HAL_SUCESS) {
		ret = MBEDTLS_ERR_RSA_UNSUPPORTED_OPERATION;
	}

FREE:
	sl_deinit(shnd);
	return ret;
```